python/lsst/ctrl/execute/slurmPlugin.py

```diff
@@ -188,6 +187,180 @@ def createAllocationFile(self, input):
+                    shash = hash[:6]
+                    empty_list = []
+                    label_dict[job_label] = empty_list
+
+                # Loop over the Large jobs and categorize
+                for jid in list(condorq_bps_large.keys()):
+                    ajob = condorq_bps_large[jid]
+                    this_label = ajob["bps_job_label"]
+                    this_list = label_dict[this_label]
+                    this_list.append(ajob)
+
+                for job_label in unique_labels:
+                    if verbose:
+                        print(f"\n{job_label}")
+                    existingGlideinsIdle = 0
+                    numberOfGlideinsRed = 0
+                    numberOfGlideins = 0
+                    alist = label_dict[job_label]
+                    thisMemory = alist[0]["RequestMemory"]
+                    useCores = alist[0]["RequestCpus"]
+                    if useCores < autoCPUs:
+                        useCores = autoCPUs
+                    hash = hashlib.sha1(job_label.encode("UTF-8")).hexdigest()
+                    shash = hash[:6]
+                    numberOfGlideins = len(alist)
+                    jobname = f"{auser}_{shash}"
+                    print(f"{job_label} {jobname} target {numberOfGlideins}")
+
+                    # Do not submit squeue commands rapidly
+                    time.sleep(2)
+                    # Check Slurm queue Idle Glideins
+                    batcmd = f"squeue --noheader --states=PD --name={jobname} | wc -l"
+                    if verbose:
+                        print("The squeue command is: %s " % batcmd)
+                    try:
+                        resultPD = subprocess.check_output(batcmd, shell=True)
+                    except subprocess.CalledProcessError as e:
+                        print(e.output)
+                    existingGlideinsIdle = int(resultPD.decode("UTF-8"))
+                    if verbose:
+                        print(f"existingGlideinsIdle {jobname}")
+                        print(existingGlideinsIdle)
+
+                    numberOfGlideinsRed = numberOfGlideins - existingGlideinsIdle
+                    if verbose:
+                        print(f"{job_label} {jobname} reduced {numberOfGlideinsRed}")
+
+                    cpuopt = f"--cpus-per-task {useCores}"
+                    memopt = f"--mem {thisMemory}"
+                    jobopt = f"-J {jobname}"
+                    cmd = f"sbatch {cpuopt} {memopt} {jobopt} {generatedSlurmFile}"
+                    if verbose:
+                        print(cmd)
+                    for glide in range(0, numberOfGlideinsRed):
+                        print("Submitting Large glidein %s " % glide)
+                        exitCode = self.runCommand(cmd, verbose)
+                        if exitCode != 0:
+                            print("error running %s" % cmd)
+                            sys.exit(exitCode)
```

Comment:
Can the various types of glideins use the same code (how many running and pending glideins are there, loop over the submission)?  If yes please break code into more functions to be reused.